src/components/LogIn.js

```diff
@@ -1,26 +1,104 @@
+import users from '../data/users.json'
 import React from 'react'
-import { auth }from '../reducers/logIn'
+import { getUsr } from '../reducers/logIn'
+import { getError } from "../reducers/errorMessage"
 import { connect } from 'react-redux'
-import { Link } from 'react-router-dom'
-
+import { Button, Form, Grid, Segment, Message } from 'semantic-ui-react'
 
 export default connect(
-    state => ({}),
-    dispatch => ({
-        auth: () => dispatch(auth())
-    })
+  state => ({
+    errorMessage: state.errorMessage.message
+  }),
+  dispatch => ({
+    getUsr: data => dispatch(getUsr(data)),
+    getError: data => dispatch(getError(data))
+  })
 )(
-    class LogIn extends React.Component {
+  class LogIn extends React.Component {
+    constructor(props) {
+      super(props);
+      this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    errorNoEmail = 'E-mail is required.'
+    errorNoPassword = 'Password is required.'
+    errorNoEmailAndPassword = 'E-mail and password are required.'
+    errorInvalidCredentials = 'E-mail address or password incorrect. Please try again.'
+    errorCheck = error => this.props.errorMessage === error
+
+    handleSubmit(e) {
+      e.preventDefault()
+
+      const formData = {}
+      for (const field in this.refs) {
+        formData[field] = this.refs[field].value;
+      }
+
+      const userEmailVerification = users.filter(user => user.email === formData.email)
+      const userPasswordVerification = userEmailVerification[0] && (userEmailVerification[0].password === formData.password)
+      const userTypeVerification = userPasswordVerification && userEmailVerification[0].type === 'admin'
 
+      userPasswordVerification && userTypeVerification ? this.props.getUsr(formData) : this.props.getUsr(null)
+      const displayErrorFor = ms => setTimeout(() => this.props.getError(null), ms)
+
+      const errorMessage = userEmailVerification === [] || (userEmailVerification.length > 0 && !userPasswordVerification) || !userPasswordVerification || !userTypeVerification ? this.errorInvalidCredentials : null
+      errorMessage && this.props.getError(errorMessage) && displayErrorFor(5000)
+
+      const errorMessageEmailFieldEmpty = formData.email === "" && formData.password !== "" ? this.errorNoEmail : null
+      errorMessageEmailFieldEmpty && this.props.getError(errorMessageEmailFieldEmpty) && displayErrorFor(5000)
+
+      const errorMessagePasswordFieldEmpty = formData.password === "" && formData.email !== "" ? this.errorNoPassword : null
+      errorMessagePasswordFieldEmpty && this.props.getError(errorMessagePasswordFieldEmpty) && displayErrorFor(5000)
+
+      const errorMessageEmailAndPasswordFieldEmpty = formData.password === "" && formData.email === "" ? this.errorNoEmailAndPassword : null
+      errorMessageEmailAndPasswordFieldEmpty && this.props.getError(errorMessageEmailAndPasswordFieldEmpty) && displayErrorFor(5000)
```

Comment:
Tu się bardzo dużo dzieje, zdecydowanie warto by to wydzielić do mniejszych plików.